test/unit/puppetlabs/services/jruby/jruby_puppet_service_test.clj

```diff
@@ -106,8 +105,16 @@
           (is (instance? JRubyPuppet jruby-puppet))
           (is (= 0 (jruby-protocol/free-instance-count service))))
         (is (= 1 (jruby-protocol/free-instance-count service)))
+        ;; borrow and return one more time
+        (with-jruby-puppet
+          jruby-puppet
+          service)
```

Comment:
I think it's a little funky to use the macro here and an explicit `borrow` and `return` would be better.